### PR TITLE
util/config: fix #1956 config error printing

### LINKED ISSUE #1956: un-meaningful rocksdb panic error
```
2017/06/22 11:37:34.840 tikv-server.rs:152: [INFO] toml value rocksdb.defaultcf.compression-per-level: “no:no:lz4:lz4:lz4:zstd:zstd4”
2017/06/22 11:37:34.840 tikv-server.rs:107: [ERROR] RocksDB
```

The error RocksDB shows nothing here.

### PR DESCRIPTION
**Pull request by andelf**
Fixes #1956 

PTAL @zhangjinpeng1987 @siddontang

### CODE REVIEW

File: src/util/config.rs [hunk 1]
```diff
@@ -23,53 +23,43 @@ use rocksdb::{DBCompressionType, DBRecoveryMode};
...
+    match &*tp.trim().to_lowercase() {
         "no" => Ok(DBCompressionType::DBNo),
         "snappy" => Ok(DBCompressionType::DBSnappy),
         "zlib" => Ok(DBCompressionType::DBZlib),
         "bzip2" => Ok(DBCompressionType::DBBz2),
         "lz4" => Ok(DBCompressionType::DBLz4),
         "lz4hc" => Ok(DBCompressionType::DBLz4hc),
-        _ => Err(ConfigError::RocksDB),
+        "zstd" => Ok(DBCompressionType::DBZstd),
+        v => Err(ConfigError::Value(format!("cannot parse {:?} as a compression type", v))),
```
Comment (siddontang) [inline src/util/config.rs]: "invalid compression type {:?}"

File: src/util/config.rs [hunk 2]
```diff
@@ -23,53 +23,43 @@ use rocksdb::{DBCompressionType, DBRecoveryMode};
...
-            "lz4" => result.push(DBCompressionType::DBLz4),
-            "lz4hc" => result.push(DBCompressionType::DBLz4hc),
-            "zstd" => result.push(DBCompressionType::DBZstd),
-            _ => return Err(ConfigError::RocksDB),
-        }
-    }
-
-    Ok(result)
+    tp.split(':')
+        .map(parse_rocksdb_compression)
```
Comment (BusyJay) [inline src/util/config.rs]: It won't compile. And format is broken.

Comment (andelf) [inline src/util/config.rs]: ?????

Comment (BusyJay) [inline src/util/config.rs]: I made a mistake, it should be fine here.

**Comment by siddontang** [comment]
PTAL @BusyJay

File: src/util/config.rs [hunk 3]
```diff
@@ -23,53 +23,43 @@ use rocksdb::{DBCompressionType, DBRecoveryMode};
...
+    match &*tp.trim().to_lowercase() {
         "no" => Ok(DBCompressionType::DBNo),
         "snappy" => Ok(DBCompressionType::DBSnappy),
         "zlib" => Ok(DBCompressionType::DBZlib),
         "bzip2" => Ok(DBCompressionType::DBBz2),
         "lz4" => Ok(DBCompressionType::DBLz4),
         "lz4hc" => Ok(DBCompressionType::DBLz4hc),
-        _ => Err(ConfigError::RocksDB),
+        "zstd" => Ok(DBCompressionType::DBZstd),
+        v => Err(ConfigError::Value(format!("invalid compression type {:?}", v))),
```
Comment (BusyJay) [inline src/util/config.rs]: Should use `tp` instead of `v`.

Comment (andelf) [inline src/util/config.rs]: reject.
won't fix and will wait all other `v`s in codebase.

Comment (BusyJay) [inline src/util/config.rs]: `v` is a temporary variable during processing, using `tp` is more clear and straight forward for user. You can raise a pr to clean up all the other places. But since this pr is about to fix error printing, so it's fine to fix them all in this pr too.

Comment (siddontang) [inline src/util/config.rs]: any update here? 

Maybe we can keep it for now and use another PR to clean up all?

Comment (BusyJay) [inline src/util/config.rs]: Just checked out the diff, places need to be changed are L54 and L128, all are introduced in this pr.

**Comment by siddontang** [comment]
Ping @andelf

**Comment by siddontang** [comment]
PTAL @BusyJay

**Comment by BusyJay** [comment]
LGTM
